alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionEvaluator.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.alerter;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.hawkular.metrics.alerter.ConditionExpression.Function;
+import org.hawkular.metrics.model.AvailabilityBucketPoint;
+import org.hawkular.metrics.model.BucketPoint;
+import org.hawkular.metrics.model.NumericBucketPoint;
+import org.hawkular.metrics.model.Percentile;
+import org.jboss.logging.Logger;
+
+import com.udojava.evalex.Expression;
+import com.udojava.evalex.Expression.ExpressionException;
+
+public class ConditionEvaluator {
```

Comment:
Good point, a couple of the new classes were missing class-level jdoc.  The supported Expression syntax is more fully defined at the EvalEx page, I have created class jdoc that points to it, as well as adds more info.  I also added class jdoc for ConditionExpression.